### ISSUE TITLE
Monitoring causes OOM Error when python holds more than 50% of available memory

### ISSUE THREAD

**Issue by mthrok** [openai/gym#110]
**tl;dr** It is useful if `gym` supports a recording method which does not involve `subprocess.Popen`. 

When monitoring is enabled, `ImageEncoder` execute recoding in subprocess with `Popen`.

`Popen` requests OS an extra memory as much as Python is currently eating up. (See [this StackOverflow answer](http://stackoverflow.com/a/13329386) for more detailed explanation.)

This can be troublesome when running memory-intensive training process (like deep learning).
When training cosumes more than 50 % of physical memory*, then `Popen` throws `OSError: [Errno 12] Cannot allocate memory`.

\* To avoid this, one can [create swap file](https://www.digitalocean.com/community/tutorials/how-to-add-swap-on-ubuntu-14-04), by which virtual memory size is increased.

**Comment by tlbtlbtlb** [comment]
I don't see a practical way to run a video encoder other than by forking. Instead of adding swap, you can just set a higher overcommit ratio:
```sh
# echo 90 > /proc/sys/vm/overcommit_ratio
```

**Comment by mthrok** [comment]
Oh, thanks. I will give it a try.

**Comment by mthrok** [comment]
@tlbtlbtlb 
I changed overcommit memory setting and it solved. Thanks.

```
sudo bash -c "echo vm.overcommit_memory=1 >> /etc/sysctl.conf"
sudo sysctl -p
```
